1499 - Convert MLS to use geolocate and geosubmit

crankycoder opened the pull request:
This closes off #1499 to switch to the `geolocate` and `geosubmit` APIs from `submit` and `search`.

@hannosch you've got the JSON blobs for this PR already, let me know if there's anything terrible about the JSON.

I've also got a commit in here to add an explicit UnittestLogger and DebugLogger as there was no good way to get adb logs at the debug level in a debug build on an actual device.

crankycoder committed:
* Reworked JSON serialization for geolocate and geosubmit.
* Added basic tests around JSON serialization.
* Repoint to ichnaea's geolocate and geosubmit methods from search and submit
* Updated ObservationPoint to use new count-free JSON
* Added zero length guards around wifi and cell arrays
* Updated the parsing of the Location object coming back from geolocate

crankycoder committed:
Added an explicit UnittestLogger and DebugLogger so that Log.d calls work in debug builds on devices.

crankycoder committed:
Wrapped the logging of the JSON response body for errors so that it only shows up in debug builds.

crankycoder committed:
Clarified arguments for `setSignalStrength` as per code review.

crankycoder committed:
Merge branch 'dev' into features/geolocate-geosubmit

crankycoder committed:
* Cleaned up merge conflicts with `DataStorageManagerTest`
* Dropped MLSTest entirely.  The verification for the blob being sent
  to MLS is in the StumblerBundleTest class now.
* Removed PII leak that might show up in Fennec
* Clean up imports and MPL licenses
* Got rid of extremely noisy debug logging in MainApp when tests run

hannosch commented on libraries/stumbler/src/test/java/org/mozilla/mozstumbler/service/core/http/MLSTest.java:
```diff
@@ -0,0 +1,59 @@
+    {
+        "latitude": -22.6,
+        "longitude": -43.4,
+        "radioType": "gsm",
```
move radioType from here into the cell mapping

hannosch commented on libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/DataStorageContract.java:
```diff
@@ -10,16 +10,19 @@ private DataStorageContract() {
+
+        // cell and wifi count are not persisted to disk
+        //public static final String CELL_COUNT = "cell_count";
+        //public static final String WIFI_COUNT = "wifi_count";
```
maybe remove those lines instead of commenting them out?

crankycoder committed:
* Fixup bugs where ASU and signalStrength are confused
* Updated int->float for changes related to switching from
  submit->geosubmit
* Got rid of setNeighboringCellInfo method as we can just use an
  overloaded constructor
* Replaced a System.currentTimeMillis dependency with ISystemClock so
  that tests can instrument time.
* Added null checks around DataStorageManager.getInstance() so that we
  don't NPE if the service hasn't started yet.
* Added test for QueuedCountsTracker when no DataStorageManager exists yet.
* added a cache timing test to the QueuedCountsTracker

crankycoder committed:
Split geolocate and geosubmit JSON blobs as geolocate does not include
accuracy or altitude fields

crankycoder committed:
This gets rid of the phone type (GSM or CDMA) as we no longer need it for anything.

crankycoder committed:
force the DataStorageManager to be null
